src/controller/SoldierController.ts

```diff
@@ -0,0 +1,38 @@
+import { Request, Response } from 'express';
+import Soldier from '../repository/soldierRepo';
+import soldierSchema from '../repository/schemaValidation';
+
+export async function insertSoldier(req:Request, res:Response) {
+  try {
+    const validateBody = soldierSchema.parse(req.body);
+    const soldier = new Soldier(validateBody);
+    await soldier.save((err, savedSoldier) => {
+      if (err) {
+        res.status(400).send(err);
+      } else {
+        res.status(201).send(savedSoldier);
+      }
+    });
+  } catch {
+    res.status(400).send('bad request, wrong fields');
+  }
+}
+
+export async function getSoldierByID(req:Request, res:Response) {
+  const { id } = req.params;
+  const SoldierByID = await Soldier.findById({ _id: id });
```

Comment:
SoldierById - camelCase (soldierByID)

And also, this is a function name not a variable name